### DISCUSSION TITLE
Exporting StateVariables in Exodus Output File

### DISCUSSION THREAD

**Discussion by shrituntunroy** [idaholab/moose#28199] [Q&A Modules: Solid mechanics]
Hi Moose Team,

I am a new MOOSE user and currently learning to use the software. I have a question regarding the export of state variables in the Exodus output file.

I am aware of the technique using the AuxKernel system in MOOSE, which involves defining an AuxVariable and exporting this variable with the following syntax:
```bash
  [./Ep_eff]
    type = StateVariable
    variable = Ep_eff
    sdv_id = 47
    execute_on = timestep_end
  [../]
```
However, my problem involves around 100 state variables. Defining 100 AuxVariables and 100 AuxKernels would be very cumbersome. Is there any way to export all the state variables using a more concise syntax in the Exodus output file?

Thank you for considering my query.

**Comment by GiudGiud** [comment]
Hello

You can create an action with a reduced syntax that programmatically performs this operation for you.
https://mooseframework.inl.gov/source/actions/Action.html

If you don't want to modify the code, then simply create this length input file with 100x auxkernels and separate it from the main input input.
in the main input file you can do 
```
!include other_input.i
```

Guillaume

**Comment by shrituntunroy** [comment]
Thankyou Guillaume for your help
